### ISSUE TITLE
First rule of added style ignored

### ISSUE THREAD

**Issue by al1ydn** [Tampermonkey/tampermonkey#1944]
## Expected Behavior
First rule of added style should not be ignored.

## Actual Behavior
First rule of added style ignored. I have to put an empty rule to beginning of .css file. Otherwise, my actual first rule is not working.

## Specifications

- Chromium: 120.0.6099.72
- TM: 5.0.0
- OS: Windows 10 19045.3803

## Script

```js
// ==UserScript==
// @name         IMDb-FullPoster
// @namespace    http://tampermonkey.net/
// @version      0.1
// @description  try to take over the world!
// @author       You
// @match        https://www.imdb.com/title/*
// @icon         https://www.google.com/s2/favicons?sz=64&domain=imdb.com
// @grant        GM_addStyle
// @grant        GM_getResourceText
// @resource     myCss file://C:\Users\<myusername>\Desktop\Temp\Tampermonkey\IMDb-FullPoster.css
// @require      file://C:\Users\<myusername>\Desktop\Temp\Tampermonkey\IMDb-FullPoster.js
// ==/UserScript==
```

## IMDb-FullPoster.js

```js
const myCss = GM_getResourceText("myCss");
GM_addStyle(myCss);
```

## IMDb-FullPoster.css

```css
/* ignored rule */
.class1 {
    visibility: hidden;
}

/* not ignored rule */
.class2 {
    visibility: hidden;
}
```

**Comment by derjanb** [comment]
This is working fine here with:

```CSS
/* ignored rule */
.imdb-header {
  visibility: hidden;
}

/* not ignored rule */
.ipc-page-wrapper {
  visibility: hidden;
}
```

Both the header and the page are hidden.

Please add a concrete example how to reproduce the issues.

**Comment by al1ydn** [comment]
To reproduce the issue, I created a new script and all required files. I copied all codes to new files. But I could not reproduce the issue. Everything worked as expected. All files were identical. I couldn't see any difference between files. Then I noticed there is a difference. Encoding of the files that doesn't work as expected were UTF8 with BOM. After changing their encodings to UTF8, they started to work as expected.

I don't know why they encoded with UTF8 with BOM but Tampermonkey has nothing to do with it, I guess.
